### PR TITLE
[Profiler][Translation] Logging false by default and desactivated when using the profiler

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | #23146 
| License       | MIT
| Doc PR        | todo.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggingTranslatorPass.php [hunk 1]
```diff
@@ -28,6 +28,9 @@ public function process(ContainerBuilder $container)
         if (!$container->hasAlias('logger') || !$container->hasAlias('translator')) {
             return;
         }
+        if ($container->has('profiler')) {
+            $container->hasParameter('translator.logging') && $container->setParameter('translator.logging', false);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggingTranslatorPass.php]: This `&&` is like an "if" - that's uncommon syntax in SF code base, isn't it?
the "hasParameter" check should be moved in the "if" instead.

**Comment by nicolas-grekas** [comment]
new features are for 4.1 (master) only

**Comment by Simperfit** [comment]
@nicolas-grekas comments done and PR rebased

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggingTranslatorPass.php (hunk 1 above)
Comment (Simperfit) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggingTranslatorPass.php]: it's not common in SF you are right.

**Comment by ogizanagi** [comment]
With these changes, it's not possible at all to set `framework.translator.logging` to `true` with profiler enabled, as it'll behave like if you set `false`, right?
Instead, what about updating the Fwb `Configuration` class only to normalize the value to `false` if no value was set explicitly and the profiler is enabled, `$this->debug` if profiler is disabled?

But IMHO, changing the default value to `false` probably is enough. Anyone can opt-in if they need it for whatever reason. Not sure it would be considered a BC break.

**Comment by Simperfit** [comment]
I may have misunderstood what we wanted here, I understood that we wanted to enforce it at `false` whenever the profiler is enabled. I guess we could just change the default value in here and in the recipe and it would be ok.

I don't think it would be considered as BC too.

**Comment by ogizanagi** [comment]
To clarify: from dev point of view, there is no apparent reason we want the logging + translator panel being enabled at the same time, as the info is redundant. 
But that doesn't mean we must enforce the logging to be disabled when the profiler is enabled, as you might have use cases for using the `LoggingTranslator` despite this. For instance by using it to collect missing translations and get a report at the end of the day thanks to a dedicated monolog handler using the `translation` channel. Not quite common surely, but may exist. 

The point is that in most case the info is just redundant and disabling this feature when the profiler is enabled (or at all) is a better default.

**Comment by Simperfit** [comment]
@ogizanagi changes done.

**Review by ogizanagi — APPROVED** [review]
As explained, it seems right to me to do this change, unless we want strict BC (then see https://github.com/symfony/symfony/pull/24785#issuecomment-341202499)

**Review by 20uf — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Review by dunglas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Simperfit.

**Pull request merged**